frontend/datagathering.cpp

```diff
@@ -64,13 +64,24 @@ void MainWindow::handle_result_cycles(HttpRequestWorker *worker)
 
     if (worker->error_type == QNetworkReply::NoError)
     {
-        // Get keys from json data
+        desiredHeader_ = getFields(ui_->instrumentsBox->currentText(), ui_->instrumentsBox->currentData().toString());
         auto jsonArray = worker->json_array;
         auto jsonObject = jsonArray.at(0).toObject();
         header_.clear();
         foreach (const QString &key, jsonObject.keys())
         {
-            header_.push_back(JsonTableModel::Heading({{"title", key}, {"index", key}}));
+            auto found = false;
+            for (auto i = 0; i < desiredHeader_.count(); i++)
+            {
+                if (std::get<0>(desiredHeader_[i]) == key)
+                {
+                    header_.push_back(JsonTableModel::Heading({{"title", std::get<1>(desiredHeader_[i])}, {"index", key}}));
+                    found = true;
+                    break;
+                }
+            }
+            if (!found)
+                header_.push_back(JsonTableModel::Heading({{"title", key}, {"index", key}})); // title_key = map["heading"]
```

Comment:
With a standard container you could do (untested in practice, but the principle is sound):
```suggestion
            auto it = std::find_if(desiredHeader_.begin(), desiredHeader_.end(), [key](const auto &&[data, name]) {return data == key; });
            if (it != desirederHeader_.end())
                header_.push_back(JsonTableModel::Heading({{"title", it->second}, {"index", key}}));
            else
                header_.push_back(JsonTableModel::Heading({{"title", key}, {"index", key}})); // title_key = map["heading"]
```